a3p-integration/proposals/z:acceptance/auction.test.js

```diff
@@ -0,0 +1,220 @@
+  longLivingBidSetup: {
+    name: 'long-living-bidder',
+    // This bid is placed in an earlier proposal
+    give: '80IST',
+  },
+  currentBidsSetup: {
+    user1: {
+      bidder: USER1ADDR,
+      bidderFund: {
+        value: 90000000,
+        denom: 'uist',
+      },
+      offerId: `user1-bid-${Date.now()}`,
+      give: '90IST',
+      price: 46,
+    },
+    gov3: {
+      bidder: GOV3ADDR,
+      bidderFund: {
+        value: 150000000,
+        denom: 'uist',
+      },
+      offerId: `gov3-bid-${Date.now()}`,
+      give: '150IST',
+      discount: '13',
+    },
+  },
+  bidsOutcome: {
+    longLivingBidder: {
+      payouts: {
+        Bid: 0,
+        Collateral: 1.68421,
+      },
+    },
+    user1: {
+      payouts: {
+        Bid: 0,
+        Collateral: 2.0,
+      },
+    },
+    gov3: {
+      payouts: {
+        Bid: 0,
+        Collateral: 3.448275,
+      },
+    },
+  },
+};
+
+test.before(async t => {
+  /** @type {RetryOptions} */
+  const pushPriceRetryOpts = {
+    maxRetries: 5, // arbitrary
+    retryIntervalMs: 5000, // in ms
+  };
+
+  /** @type {RetryOptions} */
+  const bankSendRetryOpts = {
+    maxRetries: 3, // arbitrary
+    retryIntervalMs: 3000, // in ms
+  };
+
+  // Get current round id
+  const round = await agoric.follow(
+    '-lF',
+    ':published.priceFeed.ATOM-USD_price_feed.latestRound',
+  );
+  t.context = {
+    roundId: parseInt(round.roundId, 10),
+    retryOpts: {
+      bankSendRetryOpts,
+      pushPriceRetryOpts,
+    },
+  };
+});
+
+test('run auction', async t => {
+  // Push the price to a point where only our bids can settle
+  await pushPricesForAuction(t, config.price);
+
+  // Wait until next round starts. Retry error message is useful for debugging
+  const { retryOptions, nextStartTime } =
+    await calculateRetryUntilNextStartTime();
+  await retryUntilCondition(
+    () => Promise.resolve(Date.now()),
+    res => res >= nextStartTime * 1000,
+    'next auction round not started yet [AUCTION TEST]',
+    {
+      log: t.log,
+      ...ambientAuthority,
+      ...retryOptions,
+    },
+  );
+
+  // Make sure depositor and bidders have enough balance
+  await fundAccts(t, config.depositor, config.currentBidsSetup);
+  const bidsP = placeBids(t, config.currentBidsSetup);
+  const proceedsP = depositCollateral(t, config.depositor);
+
+  // Make sure price captured
+  // We have to check this after bids are placed and collateral deposited
+  // because of https://github.com/Agoric/BytePitchPartnerEng/issues/31
+  await checkPriceCaptured('book0', scale6(config.price).toString());
+
+  // Resolves when auction finalizes and depositor gets payouts
+  const [longLivingBidderAddr] = await Promise.all([
+    getUser(config.longLivingBidSetup.name),
+    ...bidsP,
+    proceedsP,
+  ]);
+
+  // Query wallets of the actors involved for assertions
+  const [gov1Results, longLivingBidResults, user1Results, gov3Results, brands] =
+    await Promise.all([
+      agoric
+        .follow(
```

Comment:
I lament that we still have to go out through the CLI to get this functionality that is in JS: https://github.com/Agoric/agoric-sdk/blob/33309e0f2d0aa9f6a8093a0398935b4a98704943/packages/agoric-cli/src/follow.js#L104

https://github.com/Agoric/agoric-3-proposals/issues/98 is related. 

Filed https://github.com/Agoric/agoric-sdk/issues/10369

Reply:
Definitely agree